auto_submit/test/service/validation_service_test.dart

```diff
@@ -268,6 +269,183 @@ void main() {
       // We acknowledge the issue.
       assert(pubsub.messagesQueue.isEmpty);
     });
+
+    test('Exhaust retries on merge on retryable error.', () async {
```

Comment:
Can we add a test which validates the retry? First fails, and then succeeds?

Reply:
This may be difficult but I will give it a shot.

Reply:
Done.

Reply:
F.Y.I. some alternatives returning different values at different iterations: https://github.com/flutter/cocoon/blob/main/app_dart/test/foundation/utils_test.dart#L67